lib/compiler/component.ts

```diff
@@ -0,0 +1,498 @@
+import {
+  ClassDeclaration,
+  InterfaceDeclaration,
+  SyntaxKind,
+  TypeGuards
+} from 'ts-morph';
+import { Container } from './container';
+import { Unit } from './unit';
+import {
+  findDeclarationForIdentifier,
+  getBooleanDecoratorProperty,
+  getDecoratorPropertyInitializer,
+  getStringDecoratorProperty
+} from './util';
+
+export class Component {
+  private static SymbolId = 0;
+
+  private static readonly registry: {
+    container: Container<any>;
+    component: Component;
+  }[] = [];
+
+  public get name(): string {
+    if (TypeGuards.isInterfaceDeclaration(this.node)) {
+      return this.node.getName();
+    }
+    return this.node.getNameOrThrow();
+  }
+
+  private readonly symbolId = Component.SymbolId++;
```

Comment:
That should be no problem since its static to this class. A different version (file in the node_modules folder) will be a different class instance and have its own static counter.
In essence, this symbolId is used to create a serializable unique id. Using `uuid` will introduce a new (probably big) dependency and does not make this id more readable.